GitUI/UserControls/BlameControl.cs

```diff
@@ -332,14 +345,46 @@ private void ProcessBlame(string filename, GitRevision revision, IReadOnlyList<O
             var lineBuilder = new StringBuilder(lineLength + 2);
             var gutter = new StringBuilder(capacity: lineBuilder.Capacity * _blame.Lines.Count);
             var emptyLine = new string(' ', lineLength);
-            foreach (var line in _blame.Lines)
+            var cacheAvatars = new Dictionary<string, Image>();
+            var noAuthorTask = (Image)new Bitmap(Images.User80, avatarSize, avatarSize);
+            for (var index = 0; index < _blame.Lines.Count; index++)
             {
+                var line = _blame.Lines[index];
                 if (line.Commit == lastCommit)
                 {
+                    if (gitBlameDisplays != null)
+                    {
+                        gitBlameDisplays[index].Avatar = null;
+                    }
+
                     gutter.AppendLine(emptyLine);
                 }
                 else
                 {
+                    var authorEmail = line.Commit.AuthorMail?.Trim('<', '>');
+                    if (gitBlameDisplays != null)
```

Comment:
Invert to reduce nesting, and `continue`

Reply:
same here.